Week2/hw-part1.js

```diff
@@ -0,0 +1,137 @@
+  },
+];
+
+async function queryDatabase() {
+  let inputCountryCode = '';
+  let inputRegion = '';
+  let inputLanguage = '';
+  let inputRegOf = '';
+  let inputLangOf = '';
+
+  try {
+    const onSubmit = (prompt, response) => console.log(`You typed ${colors.rainbow(response)}`);
+    const response = await prompts(questions, { onSubmit });
+
+    inputCountryCode = response.CountryCode.toUpperCase();
+    inputRegion = response.region
+      .toLowerCase()
+      .split(' ')
+      .map(s => s.charAt(0).toUpperCase() + s.substring(1))
+      .join(' ');
+    inputLanguage = response.language.replace(/^\w/, c => c.toUpperCase());
+    inputRegOf = response.region2
+      .toLowerCase()
+      .split(' ')
+      .map(s => s.charAt(0).toUpperCase() + s.substring(1))
+      .join(' ');
+    inputLangOf = response.language2.replace(/^\w/, c => c.toUpperCase());
+
+    const selectCapital =      'SELECT name FROM city WHERE ID =(SELECT capital FROM country WHERE Code = ?);';
+    const selectLang =      'SELECT Language FROM countrylanguage JOIN country ON CountryCode = Code WHERE region = ?';
+    const selectNum = 'SELECT count(countryCode) FROM countrylanguage WHERE Language = ?';
+    const selectOfficial = `SELECT countryCode, name AS Country, isOfficial FROM country JOIN countryLanguage ON countryCode = Code WHERE region = '${inputRegOf}' AND language = '${inputLangOf}'`;
+    const selectContinent = `SELECT country.continent, COUNT(DISTINCT countrylanguage.language) as 'Number of Languages'
+    FROM country INNER JOIN countrylanguage 
+    ON countrylanguage.countrycode = country.code 
+    GROUP BY country.continent;`;
+
+    connection.connect();
+
+    const results1 = await execQuery(selectCapital, inputCountryCode);
+    const results2 = await execQuery(selectLang, inputRegion);
+    const results3 = await execQuery(selectNum, inputLanguage);
+    const results4 = await execQuery(selectOfficial);
+    const results5 = await execQuery(selectContinent);
+
+    // Query 1
+    console.log(`Q1. The capital is: ${results1[0].name.magenta}`.green);
+
+    // Query 2
+    console.log(`Q2. Languages spoken in ${inputRegion.magenta}`.green);
+    const getLang = Object.entries(results2);
+
+    for (i = 0; i < getLang.length; i++) {
+      console.log(getLang[i][1].Language);
+    }
+
+    // Query 3
+    console.log(
+      `Q3. Number of cities ${inputLanguage} is spoken: ${colors.magenta(
+        results3[0]['count(countryCode)'],
+      )}`.green,
+    );
+
+    // Query 4
+    console.table(
+      `Q4. ${inputLangOf} is official language in the region ${inputRegOf.magenta}: `.green,
+    );
+    for (i = 0; i < results4.length; i++) {
+      if (results4[i].isOfficial == 'T') {
+        console.log(`Official in: ${results4[i].Country}`);
+      } else {
+        console.log('FALSE');
+      }
+    }
+
+    // Query 5
+    console.log('Q5. Continents and number of languages spoken:'.green);
+    console.table(results5);
```

Comment:
You forgot to print the continent and numbers here :)

Reply:
I actually wanted to display it as a table. I guess that was not the intention of the homework. Now I changed it.